client/src/Editor/CreateObject.jsx

```diff
@@ -0,0 +1,170 @@
+    }
+
+    function handleNoParentChange(event) {
+        setNoParent(event.target.checked);
+        if (allowChangeParentPid && event.target.checked) {
+            setParent("");
+        }
+    }
+
+    function handleParentChange(event) {
+        setParent(event.target.value);
+        if (allowNoParentPid) {
+            setNoParent(event.target.value.length === 0);
+        }
+    }
+
+    function handleStateChange(event) {
+        setState(event.target.value);
+    }
+
+    function handleTitleChange(event) {
+        setTitle(event.target.value);
+    }
+
+    let categories = [];
+    for (let category in models) {
+        let children = [];
+        for (let model in models[category]) {
+            let value = models[category][model];
+            children.push(<TreeItem key={model} nodeId={value} label={model} />);
+        }
+        categories.push(
+            <TreeItem key={category} nodeId={`__${category}`} label={category}>
+                {children}
+            </TreeItem>
+        );
+    }
+    let stateControls = [];
+    states.forEach((currentState) => {
+        stateControls.push(
+            <label key={currentState + "-label"}>
+                {currentState}
+                <input
+                    type="radio"
+                    name="state"
+                    value={currentState}
+                    checked={state === currentState}
+                    onChange={handleStateChange}
+                />
+            </label>
+        );
+    });
+    let noParentControl = "";
+    if (allowNoParentPid) {
+        noParentControl = allowChangeParentPid ? (
+            <label>
+                <input type="checkbox" name="noParent" value="1" onChange={handleNoParentChange} checked={noParent} />
+                No parent PID
+            </label>
+        ) : (
+            <input type="hidden" name="noParent" value="1" />
+        );
+    }
+    let parentControl = "";
+    if (allowChangeParentPid || parent.length > 0) {
+        parentControl = allowChangeParentPid ? (
+            <label>
+                Parent ID
+                <input type="text" value={parent} name="parent" onChange={handleParentChange} required={!noParent} />
+            </label>
+        ) : (
+            <input type="hidden" value={parent} name="parent" />
+        );
+    }
+
+    return (
+        <form onSubmit={handleSubmit} className="editor__create-object">
+            {results.length > 0 ? <div>{"Results: " + results}</div> : ""}
+            <label>
+                Title
+                <input type="text" value={title} name="title" onChange={handleTitleChange} required />
+            </label>
+
+            {noParentControl}
+            {parentControl}
+            {stateControls}
+
+            <label>
+                Select Model Type:
+                <b>{selectedModel}</b>
+            </label>
+            <TreeView defaultCollapseIcon={"➖"} defaultExpandIcon={"➕"} onNodeSelect={handleSelect}>
```

Comment:
Would this have any encoding issues? Would it be better in an image file?

Reply:
I'll defer to @crhallberg on this one. :-)